nan_implementation_12_inl.h

```diff
@@ -334,7 +334,7 @@ Factory<v8::String>::New(ExternalOneByteStringResource * value) {
 
 Factory<v8::StringObject>::return_t
 Factory<v8::StringObject>::New(v8::Local<v8::String> value) {
-#if V8_MAJOR_VERSION >= 7
+#if V8_MAJOR_VERSION >= 7 || defined(NAN_HAS_V8_7_DEPRECATIONS)
   return v8::StringObject::New(v8::Isolate::GetCurrent(), value).As<v8::StringObject>();
```

Comment:
Ai, I raised this in https://github.com/nodejs/node/pull/23159#discussion_r221619113 - it means you can't distribute a pre-compiled add-on and have it work with older Node.js v10.x releases. That seems bad. Maybe we should just accept this warning as unavoidable.

Reply:
Well, for me it would mean that I would drop nan in my project. Because I know what lengths some of the users inside of Groupon will go to if they try to "fix" the warnings unfortunately. :(

Reply:
> it means you can't distribute a pre-compiled add-on and have it work with older Node.js v10.x releases.

As you mentioned in your linked comment – this would only affect people who actually *use* `StringObject::New()`, of which there aren’t many to begin with, right?

Reply:
No, I haven't seen it used much. Small consolation if you're one of the unhappy few, though.